notify/rocketchat/rocketchat.go

```diff
@@ -0,0 +1,192 @@
+	if err != nil {
+		return nil, err
+	}
+	tokenid, err := getTokenID(c)
+	if err != nil {
+		return nil, err
+	}
+	err = client.Login(&models.UserCredentials{ID: tokenid, Token: token})
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+func getTokenID(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenIDFile) > 0 {
+		content, err := os.ReadFile(c.TokenIDFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenIDFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.TokenID), nil
+}
+
+func getToken(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenFile) > 0 {
+		content, err := os.ReadFile(c.TokenFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.Token), nil
+}
+
+// Notify implements the Notifier interface.
+func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	var err error
+
+	data := notify.GetTemplateData(ctx, n.tmpl, as, n.logger)
+	tmplText := notify.TmplText(n.tmpl, data, &err)
+	if err != nil {
+		return false, err
+	}
+	titlex := tmplText(n.conf.Title)
+	if err != nil {
+		return false, err
+	}
+
+	title, truncated := notify.TruncateInRunes(titlex, maxTitleLenRunes)
+	if truncated {
+		key, err := notify.ExtractGroupKey(ctx)
+		if err != nil {
+			return false, err
+		}
+		level.Warn(n.logger).Log("msg", "Truncated title", "key", key, "max_runes", maxTitleLenRunes)
+	}
+	att := &models.Attachment{
+		Title:     title,
+		TitleLink: tmplText(n.conf.TitleLink),
+		Text:      tmplText(n.conf.Text),
+		ImageURL:  tmplText(n.conf.ImageURL),
+		ThumbURL:  tmplText(n.conf.ThumbURL),
+		Color:     tmplText(n.conf.Color),
+	}
+	numFields := len(n.conf.Fields)
+	if numFields > 0 {
+		fields := make([]models.AttachmentField, numFields)
+		for index, field := range n.conf.Fields {
+			// Rebuild the field by executing any templates and setting the new value for short
+			fields[index] = models.AttachmentField{
+				Title: tmplText(field.Title),
+				Value: tmplText(field.Value),
+				Short: field.Short,
+			}
+		}
+		att.Fields = fields
+	}
+
+	numActions := len(n.conf.Actions)
+	if numActions > 0 {
+		actions := make([]models.AttachmentAction, numActions)
+		for index, action := range n.conf.Actions {
+			rocketchatAction := models.AttachmentAction{
+				Type: "button", // Only button type is supported
+				Text: tmplText(action.Text),
+				Url:  tmplText(action.Url),
+				Msg:  tmplText(action.Msg),
+			}
+
+			actions[index] = rocketchatAction
+		}
+		att.Actions = actions
+	}
+
+	req := &models.PostMessage{
+		Channel:     n.conf.Channel,
+		Text:        tmplText(n.conf.Text),
+		Alias:       tmplText(n.conf.Alias),
+		Emoji:       tmplText(n.conf.Emoji),
+		Avatar:      tmplText(n.conf.IconURL),
+		Attachments: []models.Attachment{*att},
+	}
+
+	response, err := n.client.PostMessage(req)
```

Comment:
As mentioned in a previous comment, might have to change use of the SDK and use `notify.PostJSON` like the others.